### ISSUE TITLE
TypeError - unexpected keyword argument 'ordered'

### ISSUE THREAD

**Issue by Wesmania** [marshmallow-code/marshmallow#597]
When running our tests with marshmallow-2.13.2 and marshmallow-jsonapi-0.10.2, we encountered this stack trace:

```
tests/fa/test_updater.py:3: in <module>
    from fa import updater
src/fa/__init__.py:15: in <module>
    from game_process import instance as instance
src/fa/game_process.py:8: in <module>
    import util
src/util/__init__.py:718: in <module>
    from crash import CrashDialog
src/util/crash.py:11: in <module>
    from faf.api import BugReportSchema
/usr/lib64/python2.7/site-packages/faf/api/__init__.py:4: in <module>
    from faf.api.leaderboard_schema import LeaderboardSchema
/usr/lib64/python2.7/site-packages/faf/api/leaderboard_schema.py:1: in <module>
    from marshmallow_jsonapi import Schema, fields
/usr/lib64/python2.7/site-packages/marshmallow_jsonapi/__init__.py:2: in <module>
    from .schema import Schema, SchemaOpts
/usr/lib64/python2.7/site-packages/marshmallow_jsonapi/schema.py:31: in <module>
    class Schema(ma.Schema):
/usr/lib64/python2.7/site-packages/marshmallow/schema.py:106: in __new__
    klass.opts = klass.OPTIONS_CLASS(meta, ordered=ordered)
E   TypeError: Error when calling the metaclass bases
E       __init__() got an unexpected keyword argument 'ordered'
```

This might be related to #593.

**Comment by sloria** [comment]
Thanks for reporting. This error is certainly related to #593, which added the `ordered` argument to OPTIONS_CLASS. 

Looks like we'll need to find an alternate solution that doesn't break compatibility with 2.x. @frol Would you be willing to look into this?

**Comment by frol** [comment]
@sloria We can update the opts instance setting the `ordered` attribute in the `__new__` instead of passing it through `__init__`.

**Comment by sloria** [comment]
@frol Sounds like the right approach.

**Comment by sloria** [comment]
@frol Are you working on a PR? If you don't have time right now, let me know. I can look into it when I get home later today.

**Comment by frol** [comment]
@sloria I am working on PR. I am sorry for the breakage :(

**Comment by sloria** [comment]
No worries. Thanks for working on this!

**Comment by anapaulagomes** [comment]
Still not working. I took a look at the source code and it seems fixed. But when I did install with `pip` and even with the `.zip` file I face the same problem. :( I'm using Python 3.6.1.

**Comment by frol** [comment]
@anapaulagomes Which branch do you select for the `.zip`? `dev` branch has this breaking change since it is 3.0 branch. Either add `ordered=False` keyword argument or `**kwargs` to your custom `SchemaOpts.__init__`, that will resolve the issue and will work in both Marshmallow 2.x and 3.x. Otherwise, ensure that you use 2.x releases.

**Comment by anapaulagomes** [comment]
I selected the `2.x-line`. I tried adding `ordered=False` as the commit fixing this problem did but it did not work.

**Comment by frol** [comment]
Post the exact error you get.

**Comment by milahu** [comment]
> Still not working

probably a user error

since version 3, this should be fixed with 4a4d8ef27f3520ad06411d3f125f8f6788756983

```diff
-        klass.opts = klass.OPTIONS_CLASS(meta)
+        klass.opts = klass.OPTIONS_CLASS(meta, ordered=ordered)



 class SchemaOpts(object):
     """class Meta options for the :class:`Schema`. Defines defaults."""

-    def __init__(self, meta):
+    def __init__(self, meta, ordered=False):
```

with portia_server 2.0.8 (old buggy code) and marshmallow 3.15.0

```
  File "/nix/store/qxckzhi8j1f4m5xgk642xvkil1q40h43-portia_server-2.0.8/opt/portia_server/portia_orm/serializers.py", line 33, in <module>
    class FileSerializer(schema.Schema):
  File "/nix/store/q2h0b9cx6sambfmrdc4mwbhni2cjxsyx-python3.9-marshmallow-3.15.0/lib/python3.9/site-packages/marshmallow/schema.py", line 114, in __new__
    klass.opts = klass.OPTIONS_CLASS(meta, ordered=ordered)
TypeError: __init__() got an unexpected keyword argument 'ordered'
```


```py
# portia_server/portia_orm/serializers.py
from marshmallow import schema

class FileSerializer(schema.Schema): # line 33
    OPTIONS_CLASS = FileSerializerOpts

    def __init__(self, *args, **kwargs):
        super(FileSerializer, self).__init__(*args, **kwargs)
        # ...
```

debug ...

```py
# marshmallow/schema.py
        print("klass.OPTIONS_CLASS:")
        print(klass.OPTIONS_CLASS) # <class 'portia_orm.serializers.FileSerializerOpts'>

        # FIXME TypeError: __init__() got an unexpected keyword argument 'ordered'
        klass.opts = klass.OPTIONS_CLASS(meta, ordered=ordered)
```

so the problem is my `__init__` function

```py
# portia_server/portia_orm/serializers.py
class FileSerializerOpts(schema.SchemaOpts):
    def __init__(self, meta):
        self.ordered = True
```

fixed

```py
# portia_server/portia_orm/serializers.py
class FileSerializerOpts(schema.SchemaOpts):
    def __init__(self, meta, ordered=True):
        self.ordered = ordered
```

the error from marshmallow could be a bit more user-friendly
by reporting the class name of the broken `__init__` function
in my case `portia_orm.serializers.FileSerializerOpts`
